### PR TITLE
feat: restore lost markup tool

### PR DESCRIPTION
**Pull request by aonnikov**

### CODE REVIEW

**Review by lexiv0re — DISMISSED** [review]

File: dev/tool/src/markup.ts [hunk 1]
```diff
@@ -212,3 +220,123 @@ async function processMigrateMarkupFor (
...
+            } else if (innerTx._class === core.class.TxUpdateDoc) {
+              const updateTex = innerTx as TxUpdateDoc<Doc>
+              markup = (updateTex.operations as any)[attr.name]
+            } else {
+              continue
+            }
+
+            if (markup === undefined || !markup.startsWith('{')) continue
+
+            console.log(doc._class, doc._id, attr.name, markup)
```
Comment (kristina-fefelova) [inline dev/tool/src/markup.ts:298]: Do we need this and below logs?

Comment (aonnikov) [reply dev/tool/src/markup.ts:298]: Yeah, this is to understand what we are going to changes. So the idea is to run one time with ‘show’ parameter and see what will be changed, then you run once again with ‘restore’ and see what changed.

File: dev/tool/src/markup.ts [hunk 2]
```diff
@@ -212,3 +220,123 @@ async function processMigrateMarkupFor (
...
+
+          // try to restore by txes
+          // we need last tx that modified the attribute
+
+          const txes = await connection.findAll(core.class.Tx, query, { sort: { modifiedOn: SortingOrder.Descending } })
+          for (const tx of txes) {
+            const innerTx = TxProcessor.extractTx(tx)
+
+            let markup: string | undefined
+            if (innerTx._class === core.class.TxMixin) {
```
Comment (kristina-fefelova) [inline dev/tool/src/markup.ts:282]: Do  you need this `if`?
you didnt request  `core.class.TxMixin` : `class: { $in: [core.class.TxCreateDoc, core.class.TxUpdateDoc] }`

Comment (aonnikov) [reply dev/tool/src/markup.ts:282]: It is a bug, we need to request `core.class.TxMixin` as well

File: dev/tool/src/markup.ts (hunk 1 above)
Comment (kristina-fefelova) [reply dev/tool/src/markup.ts:298]: ok, if it's planned
I just thought that is too many logs for each tx

File: dev/tool/src/markup.ts (hunk 2 above)
Comment (aonnikov) [reply dev/tool/src/markup.ts:282]: Fixed

**Review by kristina-fefelova — APPROVED** [review]

**Pull request merged**